dali/test/python/test_operator_constant.py

```diff
@@ -0,0 +1,144 @@
+        return [
+            types.Constant(device = device, value = (1.25,2.5,3)),
+            types.Constant(device = device, value = np.array([[[1,2]],[[3,4]]], dtype=np.int32)),
+            types.Constant(device = device, value = np.array([0,1,2,3,4], dtype=np.uint8)),
+            types.Constant(device = device, value = 5.5, shape=(100,100)),
+            types.Constant(device = device, value = -4, shape=(10,20))
+        ]
+
+
+class ScalarConstantPipeline(Pipeline):
+    def __init__(self, device):
+        super(ScalarConstantPipeline, self).__init__(10, 3, device_id = 0, exec_async=True, exec_pipelined=True)
+        self.device = device
+
+    def define_graph(self):
+        device = self.device
+        return [
+            # no-op
+            ops.Reshape(device = device, shape=[1])(types.Constant(1.25)),
+            # flatten with reshape op
+            ops.Reshape(device = device)(types.Constant(np.array([[1,2],[3,4]],
+                                                        dtype = np.uint16),
+                                                        device = device),
+                                         shape = types.Constant(4))
+        ]
+
+def check(a1, a2):
+    assert(a1.dtype == a2.dtype)
+    assert(np.array_equal(a1, a2))
+
+
+ref = [
+    np.array([1.25, 2.5, 3], dtype=np.float32),
+    np.array([[[1,2]],[[3,4]]], dtype=np.int32),
+    np.array([0,1,2,3,4], dtype=np.uint8),
+    np.full([100, 100], 5.5, dtype=np.float32),
+    np.full([10, 20], -4, dtype=np.int32)
+]
+
+
+def _test_op(device):
+    pipe = ConstantPipeline(device)
+    pipe.build()
+    for iter in range(3):
+        out = pipe.run()
+        if device == "gpu":
+            out = [o.as_cpu() for o in out]
+
+        for o in range(len(ref)):
+            for i in range(len(out[o])):
+                check(out[o].at(i), ref[o])
+
+def _test_func(device):
+    pipe = ConstantFnPipeline(device)
+    pipe.build()
+    for iter in range(3):
+        out = pipe.run()
+        if device == "gpu":
+            out = [o.as_cpu() for o in out]
+
+        for o in range(len(ref)):
+            for i in range(len(out[o])):
+                check(out[o].at(i), ref[o])
+
+
+def _test_scalar_constant_promotion(device):
+    pipe = ScalarConstantPipeline(device)
+    pipe.build()
+    ref = [
+        np.array([1.25], dtype=np.float32),
+        np.array([1,2,3,4], dtype=np.uint16)
+    ]
+    for iter in range(3):
+        out = pipe.run()
+        if device == "gpu":
+            out = [o.as_cpu() for o in out]
+
+        for o in range(len(ref)):
+            for i in range(len(out[o])):
+                check(out[o].at(i), ref[o])
+
+
+def test_constant_op():
+    yield _test_op, "cpu"
+    yield _test_op, "gpu"
+
+def test_constant_fn():
+    yield _test_func, "cpu"
+    yield _test_func, "gpu"
+
+def test_scalar_constant_promotion():
+    yield _test_scalar_constant_promotion, "cpu"
+    yield _test_scalar_constant_promotion, "gpu"
+
+def main():
```

Comment:
Why not let nose run the test, and duplicate that. Should we maintain both the `main` and the test case definitions?

Reply:
It's much easier to filter it here for debugging purposes.